src/test/app/AMM_test.cpp

```diff
@@ -4132,6 +4132,341 @@ struct AMM_test : public jtx::AMMTest
         });
     }
 
+    void
+    testSelection()
+    {
+        testcase("Offer/Strand Selection");
+        using namespace jtx;
+        Account const ed("ed");
+        Account const gw1("gw1");
+        auto const ETH = gw1["ETH"];
+        auto const CAN = gw1["CAN"];
+
+        auto prep = [&](Env& env, auto gwRate, auto gw1Rate) {
+            fund(env, gw, {alice, carol, bob, ed}, XRP(2'000), {USD(2'000)});
+            env.fund(XRP(2'000), gw1);
+            fund(
+                env,
+                gw1,
+                {alice, carol, bob, ed},
+                {ETH(2'000), CAN(2'000)},
+                Fund::IOUOnly);
+            env(rate(gw, gwRate));
+            env(rate(gw1, gw1Rate));
+            env.close();
+        };
+
+        for (auto const& rates :
+             {std::make_pair(1.5, 1.9), std::make_pair(1.9, 1.5)})
+        {
+            // Offer Selection
+
+            // Cross-currency payment: AMM has the same spot price quality
+            // as CLOB's offer and can't generate a better quality offer.
+            // The transfer fee in this case doesn't change the CLOB quality
+            // because trIn is ignored on adjustment and trOut on payment is
+            // also ignored because ownerPaysTransferFee is false in this case.
```

Comment:
The only reason that `ownerPaysTransferFee` is false is because [the OwnerPaysFee feature is unsupported](https://github.com/gregtatcam/rippled/blob/amm-fixes/src/ripple/protocol/impl/Feature.cpp#L413). If that ever changes to `Supported::yes`, some of these tests will fail. If that's intentional, great! If not, you should explicitly create the `Env` objects with a `FeatureBitset` that explicitly removes `featureOwnerPaysFee`.

Reply:
Could you add that to the explanation?